File: _includes/SampleCode/SampleCodeCs/Samples/Graph/FeedPost.cs
using System;
using Windows.Foundation.Collections;
using Newtonsoft.Json;
using winsdkfb;
using winsdkfb.Graph;

namespace SampleCode
{
    public static partial class SampleFunc
    {
        static async void FeedPost()
        {
            // Get active session
            FBSession sess = FBSession.ActiveSession;
            // If the user is logged in
            if (sess.LoggedIn)
            {
                // Get current user
                FBUser user = sess.User;
                // Set caption, link and description parameters
                PropertySet parameters = new PropertySet();
                parameters.Add("title", "Microsoft");
                parameters.Add("link", "https://www.microsoft.com/en-us/default.aspx");
                parameters.Add("description", "Microsoft home page");
                // Add post message
                parameters.Add("message", "Posting from my Universal Windows app.");
                // Set Graph api path
                string path = "/" + user.Id + "/feed";
                var factory = new FBJsonClassFactory(s =>
                {
                    return JsonConvert.DeserializeObject<FBReturnObject>(s);
                });
                var singleValue = new FBSingleValue(path, parameters, factory);
                var result = await singleValue.PostAsync();
                if (result.Succeeded)
                {
                    var response = result.Object as FBReturnObject;
                }
                else
                {
                    // Posting failed
                }
            }
        }
    }
}

File: _includes/SampleCode/SampleCode/Samples/Graph/FeedPost.cpp
#include "pch.h"
#include "FBReturnObject.h"

using namespace Windows::Foundation::Collections;
using namespace Platform;
using namespace concurrency;
using namespace winsdkfb;
using namespace winsdkfb::Graph;


namespace SampleCode
{
    // Get active session
    void GraphFeedPost()
    {
        FBSession^ sess = FBSession::ActiveSession;
        if (sess->LoggedIn)
        {
            // Set caption, link and description parameters
            PropertySet^ parameters = ref new PropertySet();
            parameters->Insert(L"caption", L"Microsoft");
            parameters->Insert(L"link", L"https://www.microsoft.com/en-us/default.aspx");
            parameters->Insert(L"description", L"Microsoft home page");

            // Add message
            parameters->Insert(L"message", L"Posting from my Universal Windows app.");

            //Create Graph API path
            String^ graphPath = sess->User->Id + L"/feed";

            // Create a json class factory with a class (FBReturnObject class)
            // that can receive and parse the json response returned
            FBJsonClassFactory^ fact = ref new FBJsonClassFactory([](String^ JsonText) ->
                Object^
            {
                auto returnObject = ref new FBReturnObject();
                returnObject->Id = Windows::Data::Json::JsonObject::Parse(JsonText)->GetNamedString("id");
                return returnObject;
            });

            FBSingleValue^ sval = ref new FBSingleValue(graphPath, parameters, fact);
            create_task(sval->PostAsync()).then([=](FBResult^ result)
            {
                if (result->Succeeded)
                {
                    FBReturnObject^ response = static_cast<FBReturnObject ^>(result->Object);
                }
                else
                {
                    // Posting failed
                }
            });

        }
    }
}
